packages/common/src/ui-common/form/form.ts

```diff
@@ -153,6 +153,10 @@ export type ValidationOpts = {
 
 const defaultValidationOpts = { force: false };
 
+export type FormEventHandler<V> = (newValues: V, oldValues: V) => void;
```

Comment:
This doesn't look quite right - it assumes all form event handlers will have these 2 arguments. That's not really the case for "validate" for example.

Reply:
I'm fine with handling this in a later PR, FYI